src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.GlobalExtendedAttributes.Tests.cs

```diff
@@ -82,5 +82,27 @@ public void ExtractGlobalExtendedAttributesEntry_Throws()
                 Assert.Throws<InvalidOperationException>(() => entry.ExtractToFile(Path.Join(root.Path, "file"), overwrite: true));
             }
         }
+
+        [Theory]
+        [InlineData("key", "value")]
+        [InlineData("key     ", "   value    ")]
+        [InlineData("      key     ", "   value    ")]
+        [InlineData("many sla/s\\hes", "/////////////\\\\\\///////////")]
+        public void GlobalExtendedAttribute_Roundtrips(string key, string value)
+        {
+            var stream = new MemoryStream();
+            using (var writer = new TarWriter(stream, leaveOpen: true))
+            {
+                writer.WriteEntry(new PaxGlobalExtendedAttributesTarEntry(new Dictionary<string, string>() { { key, value } }));
+            }
+
+            stream.Position = 0;
+            using (var reader = new TarReader(stream))
+            {
+                PaxGlobalExtendedAttributesTarEntry entry = Assert.IsType<PaxGlobalExtendedAttributesTarEntry>(reader.GetNextEntry());
+                Assert.Equal(1, entry.GlobalExtendedAttributes.Count);
+                Assert.Equal(KeyValuePair.Create(key.TrimStart(), value), entry.GlobalExtendedAttributes.First());
+            }
+        }
     }
 }
```

Comment:
Can we also have a similar test in `tests/TarReader/TarReader.File.Tests.cs` but for a `PaxTarEntry`?

```cs
        [Theory]
        [InlineData("key", "value")]
        [InlineData("key     ", "   value    ")]
        [InlineData("      key     ", "   value    ")]
        [InlineData("many sla/s\\hes", "/////////////\\\\\\///////////")]
        public void ExtendedAttribute_Roundtrips(string key, string value)
        {
            var stream = new MemoryStream();
            using (var writer = new TarWriter(stream, leaveOpen: true))
            {
                writer.WriteEntry(new PaxTarEntry(TarEntryType.Directory, "entryName", new Dictionary<string, string>() { { key, value } }));
            }

            stream.Position = 0;
            using (var reader = new TarReader(stream))
            {
                PaxTarEntry entry = Assert.IsType<PaxTarEntry>(reader.GetNextEntry());
                Assert.Equal(1, entry.ExtendedAttributes.Count);
                Assert.Equal(KeyValuePair.Create(key.TrimStart(), value), entry.ExtendedAttributes.First());
            }
        }
```

Reply:
It would be nice to have a test that reflects what the person intended to do.
Have two entries with colliding standard names but with different names specified on the "path" extended attribute. 
1. Verify ExtractToDirectory(..., overwriteExistingFiles: false) does not fail.
2. Verify that the extracted files honor the extended attribute and not the standard one.

Reply:
I'd encourage you to submit a PR that adds that.  The purpose of this PR is fixing the parsing of that particular field.  If after this your suggested test doesn't pass, there's another issue at play.